src/displayBackend/DisplayBackend.cpp

```diff
@@ -100,9 +100,19 @@ void DisplayFrontendHandler::onBind()
 	{
 		LOG(mLog, DEBUG) << "Found connector: " << conIndex;
 
-		createConnector(conBasePath + to_string(conIndex) + "/",
-						conIndex, buffersStorage);
-
+        try 
+		{
+			createConnector(conBasePath + to_string(conIndex) + "/",
+							conIndex, buffersStorage);
+		}
+		catch(const std::exception& err)
```

Comment:
Looks like you are just "eating" the error. I'm not sure that this is the right thing to do.

Reply:
The call of onBind is coming from FrontendHandlerBase::onStateConnected.
Any way the exception has to be 'eaten' because this is the event handler.
The code looks like
if (mBackendState == XenbusStateInitialising ||
                mBackendState == XenbusStateInitWait)
        {
                onBind();

                setBackendState(XenbusStateConnected);
        }

so, it is risky to allow the exception goes outside the class - unpredictable behavior.
But, maybe reasonable to handle it inside of FrontendHandlerBase::onStateConnected and do not change the status at XenbusStateConnected. Here I need  a help to understand the workflow, but anyway, exception will be 'eaten' and just a thing we can do - put this info in the log.

Reply:
I strongly disagree there. You are changing program flow, but as you admitted, you don't fully understand it.

For instance, this would leave front-end in "half connected" state and will effectively pause the boot process.

Current situation is quite good: if there is a problem, back-end would crash with easily debuggable coredump. But with your change back-end will pretend that all is fine, leaving guests in incorrect state. You should handle such things in the proper way.

Reply:
ok, is it possible update the corresponded issue with the vision of the next steps to do ? The research to find the solution will be continued.